### PR TITLE
[Deps] [breaking] update `cheerio` to v1

### PR DESCRIPTION
**Pull request by ljharb**
See https://github.com/cheeriojs/cheerio/issues/1047#issuecomment-325221409 for context.

Hoping to merge this ASAP so it can be part of v3.

cc @jugglinmike

This semver range also will include `^1.0.0` once it's released, verified on https://semver.npmjs.com/

### CODE REVIEW

**Comment by aweary** [comment]
@ljharb test suite seems to be failing

```
TypeError: _cheerio2.default.load(...)(...).root is not a function
```

**Comment by ljharb** [comment]
Yes, I can't get them running locally yet to fix :-/

**Comment by jugglinmike** [comment]
`root` is a static method on the jQuery-like object produced by `load`. I guess
the recommendation we made is a little inaccurate. I'll think about how to make
a better explanation for our migration guide.

For your purposes here, could you try exposing simply `cheerio.load('')(html)`?

**Comment by ljharb** [comment]
What's the functional difference? Is the fragment not jquery-like?

What about `cheerio(cheerio.load('')(html))`?

**Comment by jugglinmike** [comment]
Here again, my explanation is more confusing than helpful. "jQuery-like" can
mean a couple things, depending on your interpretation of "jQuery".

The return value of `cheerio.load('')` could be called "jQuery-like" because it
returns a *function* that is like the global function defined by jQuery (i.e.
`window.$` or `window.jQuery`). This functions has the static methods defined
by jQuery such as `parseHTML`, but it also has a Cheerio-specific static method
named `root`.

The return value of `cheerio.load('')('<div>')` is "jQuery-like" because it
returns an array-like *object* whose members are elements (sometimes called a
"jQuery collection").

My apologies for making this more confusing than it needs to be. If it's any
consolation, this conversation is going to help me improve Cheerio's migration
guide, and that will hopefully make the transition easier for others when we do
publish version 1.0. I'd be happy to contribute to this branch and get things
passing, but I may not have any availability until this weekend. Please let me
know!

**Comment by ljharb** [comment]
That sounds great; I'll add you to my fork today so you can add to this PR.

File: .travis.yml [hunk 1]
```diff
@@ -3,12 +3,12 @@ node_js:
   - "8"
   - "6"
   - "4"
+before_install:
+  - 'if [ "${TRAVIS_NODE_VERSION}" = "0.6" ]; then npm install -g npm@1.3 ; elif [ "${TRAVIS_NODE_VERSION}" != "0.9" ]; then case "$(npm --version)" in 1.*) npm install -g npm@1.4.28 ;; 2.*) npm install -g npm@2 ;; esac ; fi'
```
Comment (nfcampos) [inline .travis.yml]: why do we need this and the `install` script?

File: .travis.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ sudo: false
 matrix:
   fast_finish: true
   include:
-   - node_js: "6"
+   - node_js: "node"
```
Comment (nfcampos) [inline .travis.yml]: why this change?

Comment (ljharb) [reply .travis.yml]: There's no benefit to pinning the linter on node 6.

Comment (nfcampos) [reply .travis.yml]: is there a benefit to pinning the karma tests, seeing as they run in a browser?

File: .travis.yml (hunk 1 above)
Comment (ljharb) [reply .travis.yml]: It ensures npm doesn't break, and deps can be installed on any node version. This will be rebased out tho; only the latest commit on the PR matters.

Comment (nfcampos) [reply .travis.yml]: but we dont use versions of node in travis other than 8, 6 and 4, so why test for 0.6 and 0.4, etc?

Comment (ljharb) [reply .travis.yml]: Because then I can copy paste my known-correct Travis config lines from all my other repos :-)

File: .travis.yml (hunk 2 above)
Comment (ljharb) [reply .travis.yml]: Only if the karma tools are c deps. I don't know much about karma.

**Comment by ljharb** [comment]
OK, got this working :-)

The effective breaking change is that now `render` returns a wrapper around the *rendered thing*, just like shallow and mount will (once v3 is done) - instead of a “root” that requires you dive into `.children()`. This will make all our APIs more consistent.

**Comment by jugglinmike** [comment]
I discovered the same thing last night, but I didn't have the time/imagination
to think of a workaround to preserve Enzyme's current behavior. Even now, I'm
not sure it's possible--that old root "element" was a little like a document
fragment in that it had children but no representation in the rendered DOM.

...so I'm glad that not only is the breaking change acceptable, but that you
find it preferable for consistency's sake!

On the Cheerio side, I plan on updating the migration guide with information
about the `root` method. We didn't modify its implementation, but now that
we're working with more realistic documents, users should be aware that what
they get back will be different than prior releases (though more in-line with
the browser overall).

**Comment by nfcampos** [comment]
@ljharb are we compiling a list of breaking changes for v3?

File: .travis.yml (hunk 2 above)
Comment (nfcampos) [reply .travis.yml]: i dont think it makesany difference, but no point in changing it now

**Comment by ljharb** [comment]
@nfcampos i believe leland has one in progress.
